concepts/40 Angular Components/40 Common Features/10 Tree Shaking.md

```diff
@@ -10,4 +12,21 @@ If tree shaking is not configured in your application, import exports from speci
     <!--TypeScript-->
     import { DxButtonModule } from 'devextreme-angular/ui/button';
 
-[tags] angular
\ No newline at end of file
+Please note that the tree shaking mechanism works only when you create the **production** bundle.
+
+## Bundle Size
+
+DevExtreme code can be divided to three main groups:
+1) Utilities Code.
+2) Data Visualization Components Code.
+3) Ui Components Code.
+
+Utilities Code is the code that is used by all our components. This code is always imported to the application when you add a DevExtreme component. It can't be removed by the Tree Shaking mechanim as our components won't work without it. An example of such code is our Option Manager or Events Engine.
+
+Data Visualization components code is independent. This means that you can import the Chart module and this will add only the Chart component code to the bundle. However, the Chart component is quite complex, so it will affect the size of the resulting bundle.
+
+The code of UI Components mostly independent. This means that you can import a simple component (such as dxButton) and the resulting bundle will contain only this component code. However, complex components, such as dxDataGrid, depend on multiple simple components. So, if you import dxDataGrid to the application, the resulting bundle will contain the dxScrollView, dxButton, dxSelectBox, dxTextBox and other components code.
```

Comment:
just a minor suggestion to add a couple of popular components  here 
such as dxDataGrid -> such as dxDataGrid, dxScheduler, dxPivotGrid, etc. , include simple components...

So, if you import dxDataGrid to the application, the resulting bundle will contain the dxScrollView, dxButton, dxSelectBox, **dxForm**, **dxPopup** and other smaller components.

Reply:
Fixed